pages/shared/[folderId].tsx

```diff
@@ -0,0 +1,73 @@
+import { useGetFolder } from "@/src/folder/data-access-folder";
+import { Layout } from "@/src/sharing/feature-layout";
+import { SharedLayout } from "@/src/page-layout/SharedLayout";
+import { CardList } from "@/src/link/ui-card-list";
+import { FolderInfo } from "@/src/folder/ui-folder-info";
+import { ReadOnlyCard } from "@/src/link/ui-read-only-card";
+import { SearchBar } from "@/src/link/ui-search-bar";
+import { useSearchLink } from "@/src/link/util-search-link/useSearchLink";
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+interface UserData {
+  id: number;
+  name: string;
+  email: string;
+  image_source: string;
+}
+
+const SharedPage = () => {
+  const { data } = useGetFolder();
+  const { profileImage, ownerName, folderName, links } = data || {};
+  const { searchValue, handleChange, handleCloseClick, result } =
+    useSearchLink(links);
+
+  const [userData, setUserData] = useState<UserData | null>(null);
```

Comment:
API 보니까 User 객체들로 이루어진 배열이 응답되고 있네요. 그런데 제 생각엔 결국 하나의 User만 올 것 같아서, 배열로 감싸서 올 필요는 없어 보여서 응답값 타입이 이게 맞는지 의심은 되네요.
일단 응답값 타입을 그대로 신뢰해서 UI 개발을 한다면, 아래와 같이 일단 userList에 대한 타입을 `UserData[]` 처럼 `[]` 를 뒤에 붙여서 배열의 요소 타입으로 UserData를 갖는 배열임을 선언해주세요.
```suggestion
	const [userList, setUserList] = useState<UserData[] | null>(null);
```